### PR TITLE
bpo-39426: Fix outdated default and highest protocols in docs

### PR DESCRIPTION
**Pull request by mdickinson**
Some portions of the pickle documentation hadn't been updated for the pickle protocol changes in Python 3.8 (new protocol 5, default protocol 4). This PR fixes those docs.

https://bugs.python.org/issue39426

Automerge-Triggered-By: @pitrou

### CODE REVIEW

File: Lib/pickle.py [hunk 1]
```diff
@@ -409,9 +409,9 @@ def __init__(self, file, protocol=None, *, fix_imports=True,
         """This takes a binary file for writing a pickle data stream.
 
         The optional *protocol* argument tells the pickler to use the
-        given protocol; supported protocols are 0, 1, 2, 3 and 4.  The
-        default protocol is 4. It was introduced in Python 3.4, it is
-        incompatible with previous versions.
+        given protocol; supported protocols are 0, 1, 2, 3, 4 and 5.
+        The default protocol is 4. It was introduced in Python 3.4, it
```
Comment (mdickinson) [inline Lib/pickle.py]: This isn't quite grammatical. Either the comma in this line should be a full stop, or it should be a semicolon, or the last "it" on this line should be replaced with "and". I favour the last option.

Comment (ammaraskar) [reply Lib/pickle.py]: "and" here sounds good.

**Comment by mdickinson** [comment]
Whoops; I did a search and replace without noticing that I was editing the clinic files. Will fix.

**Review by ammaraskar — APPROVED** [review]

**Review by pitrou — APPROVED** [review]

**Comment by pitrou** [comment]
Thank you Mark!

**Pull request merged**